model/src/main/java/org/projectnessie/model/Namespace.java

```diff
@@ -92,8 +91,9 @@ public static Namespace of(String... elements) {
           String.format(ERROR_MSG_TEMPLATE, Arrays.toString(elements)));
     }
 
-    String name = String.join(DOT, Arrays.asList(elements));
-    return ImmutableNamespace.builder().name(name).id(name).build();
+    List<String> list = Arrays.asList(elements);
+    String name = String.join(DOT, list);
+    return ImmutableNamespace.builder().name(name).id(name).elements(list).build();
```

Comment:
Why do we store name/id/element separately? Could we not derive the others from elements?